### ISSUE TITLE
DelegateProxy crashes with UITableView in some scenarios

### ISSUE THREAD

**Issue by svenfy** [ReactiveX/RxSwift#2081]
# Description

run demo, click the first row, navigate to `Example1_CustomizationUsingTableViewDelegate`

> 2019-10-02 23:37:03.091294+0800 Example[6933:116418] [TableView] Warning once only: UITableView was told to layout its visible cells and other contents without being in the view hierarchy (the table view or one of its superviews has not been added to a window). This may cause bugs by forcing views inside the table view to load and perform layout without accurate information (e.g. table view bounds, trait collection, layout margins, safe area insets, etc), and will also cause unnecessary performance overhead due to extra layout passes. Make a symbolic breakpoint at UITableViewAlertForLayoutOutsideViewHierarchy to catch this in the debugger and see what caused this to occur, so you can avoid this action altogether if possible, or defer it until the table view has been added to a window. Table view: <UITableView: 0x7f9dd2063600; frame = (-19.5 -146.5; 414 896); clipsToBounds = YES; autoresize = RM+BM; gestureRecognizers = <NSArray: 0x600001416af0>; layer = <CALayer: 0x600001a203c0>; contentOffset: {0, 0}; contentSize: {414, 0}; adjustedContentInset: {0, 0, 0, 0}; dataSource: <RxCocoa.RxTableViewDataSourceProxy: 0x600003e606c0>>

# Advice
修改`/RxCocoa/Common/DelegateProxyType.swift`中的
```Swift
object.layoutIfNeeded()
```
为
```Swift
if let tv = object as? UITableView {
    if tv.window != nil {
        object.layoutIfNeeded()
    }
} else {
    object.layoutIfNeeded()
}
```

**Comment by freak4pc** [comment]
What demo are you referring to? @svenfy

**Comment by chiliec** [comment]
I think @svenfy means this https://github.com/RxSwiftCommunity/RxDataSources/blob/master/Examples/Example/Example1_CustomizationUsingTableViewDelegate.swift

**Comment by freak4pc** [comment]
Can you clarify the fix you posted? Is it related to UIScene in any way or its been broken this whole time?

**Comment by iNoles** [comment]
Similar problem on https://github.com/RxSwiftCommunity/RxDataSources/issues/331

**Comment by Sharath8805** [comment]
Hey, I just came across this problem. The app crashes when we deallocate a dispose bag that maintains a table datasource. The app doesn't crash on popviewcontroller but only on popToRootViewController(Could be based on my view layouts, however the problem is real).

  ```
return Disposables.create { [weak object] in
                    subscription.dispose()
                    **object?.layoutIfNeeded()**
                    unregisterDelegate.dispose()
                }
```
The above code in DelegateProxyType.swift needs to look for its superview/window. On some occasions, we call layoutSubviews when there isn't a superview during deinit.

**Comment by freak4pc** [comment]
Hey @Sharath8805, sorry for not replying sooner. Any chance you can make a quick repro project so we know we're fixing the right thing? 

Thanks!

**Comment by freak4pc** [comment]
This was already fixed, see #2149. Will be released in next version.
